src/Illuminate/Validation/Validator.php

```diff
@@ -725,6 +725,34 @@ public function getData()
         return $this->data;
     }
 
+    /**
+     * Validate the rules and get the data under validation.
+     *
+     * @return \Illuminate\Support\Collection
+     *
+     * @throws \Illuminate\Validation\ValidationException
+     */
+    public function getDataValidated()
+    {
+        $this->validate();
+
+        return $this->getDataForRules();
+    }
+
+    /**
+     * Get the data under validation only for the loaded rules.
+     *
+     * @return \Illuminate\Support\Collection
+     */
+    public function getDataForRules()
+    {
+        $ruleKeys = collect($this->getRules())->keys()->map(function ($rule) {
+            return str_contains($rule, '.') ? explode('.', $rule)[0] : $rule;
```

Comment:
Just a small optimization, but I suspect the following is faster and does the same:

```php
return explode('.', $rule, 2)[0];
```

The str_contains is not necessary, and the limit arg to explode will increase performance by early termination.

Reply:
Great -- I actually copied that line from Foundation/Providers/FoundationServiceProvider.php so the optimization can be made there as well.

Is there any way this can go into 5.6? I really need this in a project I'm working on.

Reply:
Yes, please target the 5.6 branch. I marked it as 5.7 because you targetted master. :)

Reply:
Tests would be great please. :)

Reply:
I've added tests. Anything else I can do to get this merged faster?

Reply:
TravisCI appears to be failing due to another commit that is already on 5.6

Cannot redeclare Illuminate\Tests\Database\DatabaseQueryBuilderTest::testWhereTimePostgres